.github/workflows/release.yml

```diff
@@ -0,0 +1,103 @@
+        required: false
+        type: boolean
+        default: true
+
+env:
+  RUSTFLAGS: -Dwarnings
+  RUST_BACKTRACE: 1
+  # Change to specific Rust release to pin
+  rust_stable: stable
+  rust_nightly: nightly-2024-07-07
+  # When updating this, also update relevant docs
+  rust_min: '1.81'
+  RELEASE_ROLE_ARN: ${{ secrets.RELEASE_ROLE_ARN }}
+  TOKEN_KEY_NAME: ${{ secrets.TOKEN_KEY_NAME }}
+  AWS_REGION: us-west-2
+
+permissions:
+  id-token: write  # This is required for requesting the JWT
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Rust ${{ env.rust_nightly }}
+        uses: dtolnay/rust-toolchain@stable
+        with:
+            toolchain: ${{ env.rust_stable }}
+
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-nextest
+
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Verify tag matches Cargo.toml version
+        if: ${{ inputs.dry_run }} == false
+        run: |
+          # Get latest tag from all tags
+          LATEST_TAG=$(git describe --tags --abbrev=0)
+          echo "Latest tag: $LATEST_TAG"
+
+          # Extract version from tag (remove 'v' prefix)
+          TAG_VERSION="${LATEST_TAG#v}"
+
+          # Extract version from Cargo.toml
+          CARGO_VERSION=$(grep -m 1 '^version = ' ./aws-s3-transfer-manager/Cargo.toml | sed 's/version = "\(.*\)"/\1/')
+
+          echo "Tag version: $TAG_VERSION"
+          echo "Cargo.toml version: $CARGO_VERSION"
+
+          # Verify versions match
+          if [[ "$TAG_VERSION" != "$CARGO_VERSION" ]]; then
+            echo "::error::Version mismatch! Latest Git tag ($TAG_VERSION) does not match Cargo.toml version ($CARGO_VERSION)"
+            exit 1
+          else
+            echo "✅ Version check passed: Latest Git tag matches Cargo.toml version ($TAG_VERSION)"
+          fi
+
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ env.RELEASE_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Get Cargo registry token
+        shell: bash
+        id: cargo-registry-token
+        run: |
+          CARGO_REGISTRY_TOKEN=$(aws secretsmanager get-secret-value --secret-id ${{ secrets.TOKEN_KEY_NAME }} --region ${{ env.AWS_REGION }} | jq -r ".SecretString")
+          echo "::add-mask::$CARGO_REGISTRY_TOKEN"
+          echo "token=$CARGO_REGISTRY_TOKEN" >> $GITHUB_OUTPUT
+
+      - name: Yank previous version (for testing token)
+        shell: bash
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ steps.cargo-registry-token.outputs.token }}
+        run: |
+          # This version has already been yanked. This command succeeds if we have a token with permission to yank the crate.
+          cargo yank aws-sigv4 --version 0.55.0
```

Comment:
What? Why are we trying to yank aws-sigv4? Can we just try to publish and it will work if token is good.

Reply:
https://github.com/smithy-lang/smithy-rs/blob/main/.github/workflows/release.yml#L239-L241
Make sure we are not getting some random token from other account

Reply:
>  Make sure we are not getting some random token from other account

Wouldn't cargo publish check the token and if the token is good, then publish will succeed otherwise it will fail?

Reply:
I mean if the token is from some other account (Like i can create an account and try to publish this), I assume the publish will also succeed.

Just some extra testing steps to make sure we are on the right track.

Reply:
If we publish the first version of transfer manager crate using token X, I don't think token Y will result in a successful publish unless it has the authorization. If it has the authorization to publish our crate, it will probably have the authorization to yank aws-sigv4.

Reply:
Yeah, an attempt to `aws-sigv4` version 0.55.0 is a sanity check that we have a valid crates.io token (it's an idempotent operation BTW since the crate of that version has already been yanked and doing so repeatedly is safe).
That setup makes most sense where releases occur every day and we cannot afford to fail with an invalid crates.io token; the risk of having an invalid token becomes high after we've just rotated it to a new token, and having this sanity check is useful. If we manually publish one crate, however, we can probably afford to fail with an invalid token, and simply replace it with a new one.

Also, if we source crates.io token from smithy-rs, we probably don't need to have this check, since the sanity check has already been done in on the smithy-rs side.

Reply:
Yeah, an attempt to `aws-sigv4` version 0.55.0 is a sanity check that we have a valid crates.io token (it's an idempotent operation BTW since the crate of that version has already been yanked and doing so repeatedly is safe).
That setup makes most sense where releases occur every day and we cannot afford to fail with an invalid crates.io token; the risk of having an invalid token becomes usually high after we've just rotated it to a new token, and having this sanity check is useful. If we manually publish one crate, however, we can probably afford to fail with an invalid token, and simply replace it with a new one.

Also, if we source crates.io token from smithy-rs, this check is probably not strictly necessary, since the sanity check has already been done on the smithy-rs side.

Reply:
IMHO, since it's idempotent operation, why not keep it? There is basically no cost to perform this extra check.

Reply:
Because in this particular setting, it's easier to ask for forgiveness than permission, as in Python. Especially if the token is coming from a known good source.